src/hooks/api/tokens/useRefreshToken.ts

```diff
@@ -0,0 +1,25 @@
+import { useCallback } from 'react';
+import { usePromisifiedMutation } from 'hooks/usePromisifiedMutation';
+import { graphql } from 'relay-runtime';
+import { useRefreshTokenMutation } from '__generated__/useRefreshTokenMutation.graphql';
+
+export function useRefreshToken() {
+  const [refreshTokenMutate] = usePromisifiedMutation<useRefreshTokenMutation>(graphql`
+    mutation useRefreshTokenMutation($id: DBID!) {
+      refreshToken(tokenId: $id) {
+        ... on RefreshTokenPayload {
+          token {
+            id
+          }
+        }
+      }
+    }
+  `);
+
+  return useCallback(
+    async (tokenId: string) => {
+      await refreshTokenMutate({ variables: { id: tokenId } });
+    },
+    [refreshTokenMutate]
+  );
```

Comment:
i know we don't do this often in the repo, but let's make sure to handle errors!

```js
    mutation useRefreshTokenMutation($id: DBID!) {
      refreshToken(tokenId: $id) {
        ... on RefreshTokenPayload {
          token {
            id
          }
        }
        ... on ErrInvalidInput { <-- add these
          message
        }
        ... on ErrOpenSeaRefreshFailed {  <-- add these
          message
        }
      }
    }
  `);

  return useCallback(
    async (tokenId: string) => {
      const response = await refreshTokenMutate({ variables: { id: tokenId } });

// error handle
      if (response.refreshToken?.__typename === 'ErrInvalidInput' || response.refreshToken?.__typename === 'ErrOpenSeaRefreshFailed') {
        throw new Error(`Could not refresh token: ${response.refreshToken.message}`
      }
    },
    [refreshTokenMutate]
  );
```

Reply:
Thanks for the guide robin! Learn new stuff on graphql 😆